### PR TITLE
Notification API

### PR DESCRIPTION
**Pull request by Vendicated**
TODO
- [x] implement native notifications
- [x] implement customisable position and duration
- [x] add settings ui

### CODE REVIEW

File: src/api/Notifications/styles.css [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+    border-radius: 6px;
+}
+
+/* Discord adding 3km margin to generic tags */
+.vc-notification h2 {
+    margin: unset;
+}
+
+.vc-notification-content {
+    display: flex;
```
Comment (xafn) [inline src/api/Notifications/styles.css:38]:
```suggestion
```

**Review by xafn — CHANGES_REQUESTED** [review]
guhh

**Comment by Vendicated** [comment]
Ready

File: src/api/Notifications/NotificationComponent.tsx [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+            <div className="vc-notification">
+                {icon && <img className="vc-notification-icon" src={icon} alt="" />}
+                <div className="vc-notification-content">
+                    <Forms.FormTitle tag="h2">{title}</Forms.FormTitle>
+                    <div>
+                        {richBody ?? <p className="vc-notification-p">{body}</p>}
+                    </div>
+                </div>
+            </div>
+            {image && <img className="vc-notification-img" src={image} alt="" />}
```
Comment (MeguminSama) [inline src/api/Notifications/NotificationComponent.tsx:83]: why the `alt=""`?

File: src/api/Notifications/Notifications.tsx [hunk 3]
```diff
@@ -0,0 +1,92 @@
...
+import { ReactDOM } from "@webpack/common";
+import type { ReactNode } from "react";
+import type { Root } from "react-dom/client";
+
+import NotificationComponent from "./NotificationComponent";
+
+const NotificationQueue = new Queue();
+
+let reactRoot: Root;
+let id = 42;
```
Comment (MeguminSama) [inline src/api/Notifications/Notifications.tsx:30]: Why is this 42?

File: src/components/VencordSettings/VencordTab.tsx [hunk 4]
```diff
@@ -82,52 +123,70 @@ function VencordSettings() {
...
+                    { label: "Always use Vencord notifications", value: "never" },
+                ]satisfies Array<{ value: typeof settings["notifications"]["useNative"]; } & Record<string, any>>}
+                closeOnSelect={true}
+                select={v => notifSettings.useNative = v}
+                isSelected={v => v === notifSettings.useNative}
+                serialize={identity}
+            />
+
+            <Forms.FormTitle tag="h5" className={Margins.top16}>Notification Position</Forms.FormTitle>
+            <Select
```
Comment (MeguminSama) [inline src/components/VencordSettings/VencordTab.tsx:165]: Missing padding between title and dropdown
image

**Review by MeguminSama — CHANGES_REQUESTED** [review]

File: src/api/Notifications/NotificationComponent.tsx (hunk 2 above)
Comment (Vendicated) [reply src/api/Notifications/NotificationComponent.tsx:83]: image
https://accessibility.oit.ncsu.edu/it-accessibility-at-nc-state/developers/accessibility-handbook/alternative-text/

File: src/api/Notifications/Notifications.tsx (hunk 3 above)
Comment (Vendicated) [reply src/api/Notifications/Notifications.tsx:30]: why not

File: src/components/VencordSettings/VencordTab.tsx (hunk 4 above)
Comment (Vendicated) [reply src/components/VencordSettings/VencordTab.tsx:165]: le fixed

**Review by MeguminSama — APPROVED** [review]
:shipit:

**Pull request merged**